main.py

```diff
@@ -242,20 +241,19 @@ def tmp(_, inp, out):
                 normal_outlier_count += normal_outliers_count.item()
                 w_count += quantized.weight.numel()
 
-        # upload inputs back to the device
-        inps = inps.to(layer_dev)
-        outs = outs.to(layer_dev)
-
         out_losses = []
         for j in trange(args.nsamples, desc="calc outs after quantization", leave=False):
-            outs_batch = layer(inps[j].unsqueeze(0), **forward_args)[0]
-            if not args.skip_out_loss:
-                outs_batch_loss = (outs_batch - outs[j]).float().square().view(outs_batch.shape[0], -1)\
+            if args.skip_out_loss:
+                outs[j] = layer(inps[j].to(layer_dev).unsqueeze(0), **forward_args)[0]
+            else:
+                outs_batch = layer(inps[j].to(layer_dev).unsqueeze(0), **forward_args)[0]
+                outs_batch_loss = (outs_batch - outs[j].to(layer_dev)).float().square().view(outs_batch.shape[0], -1)\
                     .mean(dim=1).sqrt()
                 outs_batch_loss /= outs_batch.view(outs_batch.shape[0], -1).float().std(dim=1)
                 out_losses.append(outs_batch_loss.item())
-            outs[j] = outs_batch
-        del outs_batch
+                outs[j] = outs_batch
+            if args.offload_activations:
```

Comment:
del outs_batch missing

Reply:
I think maybe this is not that much of compute and memory optimization

Reply:
reverted this part